### PR TITLE
Unregister JMX bean.

### PR DESCRIPTION
**Pull request by SammyVimes**
JMX bean is registered in ClusterImpl but is not unregistered. This leads to a memory leak, as JMX registry is a GC root (because of a static field that holds JMX models).

Also: 
I added resources clean up in some of the tests in ClusterTest

### CODE REVIEW

File: cluster/src/test/java/io/scalecube/cluster/ClusterTest.java [hunk 1]
```diff
@@ -12,13 +12,7 @@
...
 import java.net.InetAddress;
 import java.time.Duration;
-import java.util.ArrayList;
-import java.util.Collection;
-import java.util.Collections;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
-import java.util.Optional;
+import java.util.*;
```
Comment (artem-v) [inline cluster/src/test/java/io/scalecube/cluster/ClusterTest.java:15]: Revert back imports. Adjust your import settings.

**Review by artem-v — CHANGES_REQUESTED** [review]
Plz fix minor CR comment. Overall looks good. Thnks.

File: cluster/src/test/java/io/scalecube/cluster/ClusterTest.java (hunk 1 above)
Comment (SammyVimes) [reply cluster/src/test/java/io/scalecube/cluster/ClusterTest.java]: Thanks, fixed

**Comment by ronenhamias** [comment]
@SammyVimes thank you!

**Pull request merged**